components/allocator/tracking.rs

```diff
@@ -0,0 +1,211 @@
+            .map_or((std::ptr::null_mut(), 0), |(_, site)| {
+                (site.ptr.cast(), site.size)
+            })
+    }
+
+    pub(crate) fn note_allocation(&self, ptr: *const c_void, size: usize) {
+        if size < MIN_SIZE {
+            return;
+        }
+        IN_ALLOCATION.with(|status| status.set(true));
+        if let Some(site) = ALLOCATION_SITES.lock().unwrap().get_mut(&(ptr as usize)) {
+            site.noted = true;
+        }
+        IN_ALLOCATION.with(|status| status.set(false));
+    }
+
+    pub(crate) fn dump_unmeasured_allocations(&self) {
+        // Ensure that we ignore all allocations triggered while processing
+        // the existing allocation data.
+        IN_ALLOCATION.with(|status| status.set(true));
+        {
+            let sites = ALLOCATION_SITES.lock().unwrap();
+            let default = "???";
+            for site in sites
+                .values()
+                .filter(|site| site.size > MIN_SIZE && !site.noted)
+            {
+                let mut resolved = vec![];
+                for ip in site.frames.iter().filter(|ip| !ip.is_null()) {
+                    backtrace::resolve(*ip, |symbol| {
+                        resolved.push((
+                            symbol.filename().map(|f| f.to_owned()),
+                            symbol.lineno(),
+                            symbol.name().map(|n| format!("{}", n)),
+                        ));
+                    });
+                }
+
+                println!("---\n{}\n", site.size);
+                for (filename, line, symbol) in &resolved {
+                    let fname = filename.as_ref().map(|f| f.display().to_string());
+                    println!(
+                        "{}:{} ({})",
+                        fname.as_deref().unwrap_or(default),
+                        line.unwrap_or_default(),
+                        symbol.as_deref().unwrap_or(default),
+                    );
+                }
+            }
+        }
+        IN_ALLOCATION.with(|status| status.set(false));
+    }
+}
+
+unsafe impl<A: GlobalAlloc> GlobalAlloc for AccountingAlloc<A> {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let ptr = unsafe { self.allocator.alloc(layout) };
+        self.record_allocation(ptr, layout.size());
+        ptr
+    }
+
+    unsafe fn dealloc(&self, ptr: *mut u8, layout: Layout) {
+        unsafe { self.allocator.dealloc(ptr, layout) };
+        self.remove_allocation(ptr.cast(), layout.size());
+    }
+
+    unsafe fn alloc_zeroed(&self, layout: Layout) -> *mut u8 {
+        let ptr = unsafe { self.allocator.alloc_zeroed(layout) };
+        self.record_allocation(ptr, layout.size());
+        ptr
+    }
+
+    unsafe fn realloc(&self, ptr: *mut u8, layout: Layout, new_size: usize) -> *mut u8 {
+        self.remove_allocation(ptr.cast(), layout.size());
+        let ptr = unsafe { self.allocator.realloc(ptr, layout, new_size) };
+        self.record_allocation(ptr, new_size);
```

Comment:
Besides finding unmeasured allocations, what is the main goal of the allocation tracking? 
IMHO we would likely want to special case `realloc`. E.g. what DHAT does, is that it attributes `realloc` calls to the stack frame from the original allocation, which is useful when trying to evaulate the impact of growable datastructures like Vec, Hashmap etc. Usually you don't really care about the location where `push()` is called, but just want to know which Vec allocates heavily / frequently ...

Reply:
Finding unmeasured allocations is the only goal of this implementation. I believe there is already tooling that exists on most platforms for tasks like heap profiling.